Add README.md template (#2)
cloudposse/reference-architectures: templates/README.md

- [aws-vault](https://github.com/99designs/aws-vault)
- [chamber](https://github.com/segmentio/chamber)
- [terraform](https://www.terraform.io/)
- [kops](https://github.com/kubernetes/kops)
- [kubectl](https://kubernetes.io/docs/reference/kubectl/overview/)
- [helm](https://helm.sh/)
- [helmfile](https://github.com/roboll/helmfile)

## Layout

This repo is organized in the following way.

```text
${image_name}/
├── conf/                          # All configurations should be kept here
│   ├── Makefile                   # Makefile for controlling interactions between projects
│   ├── module1/                   # Example terraform "root" module (aka project)
│   │   └── terraform.tfvars       # Define project specific settings using a varfile (do not commit secrets)
│   ├── module2/                   # Another terraform "root" module
│   │   └── terraform.tfvars       # Terraform settings specific to this project
│   └── module3/                   # Another terraform "root" module
│       ├── file1.tf               # Overlay additional files
│       └── file2.tf               #
├── Dockerfile                     # Dockerfile that describes how to build this image

> ├── Makefile                       # Makefile that uses the `build-harness` to facilitate building the image

Reviewer: ```suggestion
├── Makefile                       # Makefile that uses the [`build-harness`](https://github.com/cloudposse/build-harness) to facilitate building the image
```

└── rootfs/                        # "Root" (`/`) filesystem which is overlayed inside of the docker image
```

## Configuration Settings

Most configuration settings are defined as environment variables. These can be set using the `ENV` declaration in the [`Dockerfile`](Dockerfile). These have been set to _sane defaults_ and shouldn't need to be touched. All these settings are required.

<details>
<summary>List of Environment Variables</summary>

| Environment Variable   | Description of the setting                                                   |
| ---------------------- | ---------------------------------------------------------------------------- |
| DOCKER_IMAGE           | _This_ docker image name. This is for the bootstrap script.                  |
| DOCKER_TAG             | The default image tag to use by the bootstrap script.                        |
| NAMESPACE              | Resource namespace used as a prefix for all AWS resources                    |
| STAGE                  | Operating stage of this account (e.g. prod, corp, audit, root)               |
| BANNER                 | Banner text to display when launching an interactive shell                   |

> | ENV MOTD_URL           | URL to a "Message of the Day" to display when launching an interactive shell |

Reviewer: ```suggestion
| MOTD_URL           | URL to a "Message of the Day" to display when launching an interactive shell |
```

> | ENV AWS_REGION         | Current operating region for this account                                    |

Reviewer: ```suggestion
| AWS_REGION         | Current operating region for this account                                    |
```

> | ENV AWS_DEFAULT_REGION | Default operating region for thid account                                    |

Reviewer: ```suggestion
| AWS_DEFAULT_REGION | Default operating region for this account                                    |
```

> | ENV AWS_ACCOUNT_ID     | AWS Account ID (used by `aws-config-setup`)                                  |

Reviewer: ```suggestion
| AWS_ACCOUNT_ID     | AWS Account ID (used by `aws-config-setup`)                                  |
```

| AWS_ROOT_ACCOUNT_ID    | AWS "Root" (parent) Account ID (used by `aws-config-setup`)                  |

> | TF_BUCKET_REGION       | Terraform state bucket                                                       |

Reviewer: ```suggestion
| TF_BUCKET       | Terraform state bucket                                                       |
```

> | TF_BUCKET              | Region where the Terraform state bucket was created                          |

Reviewer: ```suggestion
| TF_BUCKET_REGION              | Region where the Terraform state bucket was created                          |
```

| TF_DYNAMODB_TABLE     | DynamoDB table that will be used by Terraform for state locking.              |
| AWS_DEFAULT_PROFILE   | AWS Profile that will be used by `aws-vault` to assume roles.                 |
| CHAMBER_KMS_KEY_ALIAS | Default KMS key that will be used to encrypt secrets for chamber.             |

**NOTE:** You can use [`tfenv`](https://github.com/cloudposse/tfenv) to easily pass environment variables to terraform.

</details>

## Prerequisites

- [Docker](https://docs.docker.com/install/) is required to build & run all containers
- Standard development tools (e.g. `xcode-select --install` on OSX): `git`, `make`

> **NOTE:**: It should work out-of-the-box with Mac OSX, Linux, and Windows 10 (using WSL).

Reviewer: ```suggestion
**NOTE:** It should work out-of-the-box with Mac OSX, Linux, and Windows 10 (using WSL).
```

## Quick Start

Here's how to get started with this repository.

<details>
<summary>Basic Operating Instructions</summary>

### Initialize the Project

> First, let's initialize the build-harness. You only need to do this once per checkout.

Reviewer: ```suggestion
First, let's initialize the [build-harness](https://github.com/cloudposse/build-harness). You only need to do this once per checkout.
```

Reviewer: per checkout?
per account?

```bash
# Initialize the project's build-harness
make init
```

### Build Docker Image

Build the docker image we'll use for local development, to provision infrastructure or to administer AWS.

> ```

Reviewer: ```suggestion
```bash
```

> ```

Reviewer: ```suggestion
```bash
```

make docker/build
```

### Install the Wrapper Shell

Install the helper script which makes it easier to start the docker container. You only really need to do this once.

```bash
make install
```

### Run the Shell

Anytime you want to interact with tools like terraform, chamber, etc we recommend you do so from within the shell.

```bash
/usr/local/bin/${image_name}
```

**NOTE (a):** You can just run `${image_name}`, if your `PATH` contains `/usr/local/bin`
**NOTE (b):** Your `HOME` directory is mounted to `/localhost` inside of the container. This makes it easier to do local development or use your IDE of choice.

### Setup AWS IAM Account

_([inside the shell](#run-the-shell))_

Configure your AWS profile in `~/.aws/config` by running `aws-config-setup` inside of the shell. This will also prompt you to setup [`aws-vault`](https://github.com/99designs/aws-vault).

**NOTE:** You only need to do this once per AWS account.

```bash
aws-config-setup
```

### Login to AWS

_([inside the shell](#run-the-shell))_

> Run this command anytime you start a new shell and need to operate on AWS,

Reviewer: ```suggestion
Run this command anytime you start a new shell and need to operate on AWS:
```
